tools/iana_to_rflx.py

```diff
@@ -0,0 +1,221 @@
+        file.write(f"package {package_name} is\n\n")
+        for registry in root.findall(root.tag):
+            duplicates = write_registry(registry, always_valid, file, duplicates)
+        file.write(f"end {package_name};")
+
+
+def write_registry(
+    registry: Element,
+    always_valid: bool,
+    file: TextIO,
+    duplicates: List[str],
+) -> List[str]:
+    records = registry.findall("iana:record", NAMESPACE)
+    if len(records) == 0:
+        return duplicates
+    title = registry.find("iana:title", NAMESPACE)
+    if title is None or title.text is None:
+        raise IANAError("could not find registry title")
+    registry_title = _normalize_name(title.text)
+    normalized_records, duplicates = _normalize_records(records, registry_title, duplicates)
+    if not normalized_records:
+        return duplicates
+
+    file.write(f"{'':<3}type {registry_title} is\n")
+    file.write(f"{'':<6}(\n")
+    size = max((record.bit_length for record in normalized_records))
+    for record in normalized_records:
+        file.write(str(record))
+
+    file.seek(file.tell() - 3)
+    file.write(")\n")
+    file.write(f"{'':<3}with Size => {size}")
+    if always_valid and len(normalized_records).bit_length() == size:
+        file.write(", Always_Valid;")
+    else:
+        file.write(";")
+    file.write("\n\n")
+    return duplicates
+
+
+def _normalize_records(
+    records: List[Element], registry_name: str, duplicates: List[str]
+) -> Tuple[List["Record"], List[str]]:
+    normalized_records: Dict[str, Record] = {}
+    for record in records:
+        name_tag = _get_name_tag(record)
+        value_tag = "value"
+
+        name_element = record.find(f"iana:{name_tag}", NAMESPACE)
+        value_element = record.find(f"iana:{value_tag}", NAMESPACE)
+
+        if name_element is None or value_element is None:
+            continue
+
+        assert isinstance(name_element.text, str)
+        assert isinstance(value_element.text, str)
+        name = name_element.text
+        value = value_element.text
+
+        if re.search(r"RESERVED|UNASSIGNED", name, flags=re.I):
+            continue
+
+        if name in duplicates or re.match(RESERVED_WORDS, name, re.I | re.X) is not None:
+            name += f"_{value.replace('#', '_')}"
+        if name.startswith(tuple(d for d in string.digits)):
+            name = f"{registry_name}_{name}"
+        duplicates.append(name)
+
+        comment = [
+            element
+            for element in record.iterfind("*", NAMESPACE)
+            if element.tag
+            not in [f"{{{NAMESPACE['iana']}}}{name_tag}", f"{{{NAMESPACE['iana']}}}{value_tag}"]
+        ]
+
+        r = Record(name, value, comment)
+        if value in normalized_records:
+            normalized_records[value].join(r, registry_name)
+        else:
+            normalized_records[value] = r
+
+    return list(normalized_records.values()), duplicates
```

Comment:
I removed the returns in fee4364 but thought it would be better if it is explicitly returned instead of having a function with a side effect. `duplicates` is also not testable (at least i dont't know how) if never returned.

Reply:
> I removed the returns in fee4364 but thought it would be better if it is explicitly returned instead of having a function with a side effect.

The functions already had side effects before. Pure functions can be advantageous, but the question is, whether pure functions would be more suitable in this case.

> `duplicates` is also not testable (at least i dont't know how) if never returned.

Why do you want to test the local variable `duplicates`? Which case is not testable without access to `duplicates`?

Reply:
The variable does not exist anymore since efbfa5f7